packages/evm/contracts/registry/GuardiansRegistry.sol

```diff
@@ -669,6 +749,9 @@ contract GuardiansRegistry is ControlledRecoverable, IGuardiansRegistry, ERC900,
         // the activation amount since we have just added it to the available balance of the guardian.
         if (_data.toBytes4() == GuardiansRegistry(this).activate.selector) {
             _activateTokens(_guardian, _amount, _from);
+        } else if (_data.toBytes4() == GuardiansRegistry(this).lockDeactivation.selector) {
+            _activateTokens(_guardian, _amount, _from);
+            _lockDeactivation(_guardian, _from, _amount);
```

Comment:
```suggestion
            _lockDeactivation(_guardian, msg.sender, _amount);
```

Would this be more correct?

Reply:
Note that `_from` is actually the `msg.sender` in `stake` and `stakeFor`, and we are trusting the token in `receiveApproval`. Although the last will change once we support permit